core/native-bridge.ts

```diff
@@ -118,6 +118,47 @@ const convertBody = async (
   return { data: body, type: 'json' };
 };
 
+const CAPACITOR_HTTP_INTERCEPTOR = '/_capacitor_http_interceptor_';
+const CAPACITOR_HTTPS_INTERCEPTOR = '/_capacitor_https_interceptor_';
+
+// TODO: export as Cap function
+const isRelativeOrProxyUrl = (url: string | undefined): boolean =>
+  !url ||
+  !(url.startsWith('http:') || url.startsWith('https:')) ||
+  url.indexOf(CAPACITOR_HTTP_INTERCEPTOR) > -1 ||
+  url.indexOf(CAPACITOR_HTTPS_INTERCEPTOR) > -1;
+
+// TODO: export as Cap function
+const createProxyUrl = (url: string, win: WindowCapacitor): string => {
+  if (isRelativeOrProxyUrl(url)) return url;
+
+  const proxyUrl = new URL(url);
+  const isHttps = proxyUrl.protocol === 'https:';
+
+  const webviewServerUrl = new URL(
+    win.WEBVIEW_SERVER_URL ?? 'capacitor://localhost',
+  );
+  const originalHostname = proxyUrl.hostname;
+  const originalPathname = proxyUrl.pathname;
+
+  if (win.webkit?.messageHandlers?.bridge) {
+    proxyUrl.protocol = 'capacitor:';
+    if (
+      webviewServerUrl.protocol !== 'capacitor:' &&
+      webviewServerUrl.protocol !== 'http:' &&
+      webviewServerUrl.protocol !== 'https:'
+    ) {
+      proxyUrl.protocol = webviewServerUrl.protocol;
+    }
+  }
+
+  proxyUrl.hostname = webviewServerUrl.hostname;
```

Comment:
```suggestion
  let proxyUrl = new URL(url);
  const isHttps = proxyUrl.protocol === 'https:';
  const originalHostname = proxyUrl.hostname;
  const originalPathname = proxyUrl.pathname;

  proxyUrl = new URL(win.Capacitor?.getServerUrl() ?? '');
```

When using live reload on Android the replacement is not working because the server url has a port and the port is not being added into the newly created url.

I think a simple approach that also solves that issue is to recreate the proxyUrl from the capacitor server url, so you don't have to check for the hostname, port or scheme.

Reply:
Ah yeah, good suggestion 👍